script.js

```diff
@@ -97,10 +97,99 @@ class CircleMover extends Mover {
 	}
 }
 
+class Draggable {
+	constructor(x,y,w,h){
+		this.dragging = false;
+		this.rollover = false;
+		this.x = x;
+		this.y = y;
+		this.w = w;
+		this.h = h;
+		this.offsetX = 0;
+		this.offsetY = 0;
+	}
+
+	over(){
+		if (mouseX > this.x && mouseX < this.x + this.w && mouseY > this.y && mouseY < this.y + this.h) {
+			this.rollover = true;
+		} 
+		else {
+			this.rollover = false;
+		}
+	}
+
+	update(){
+		if (this.dragging) {
+			this.x = mouseX + this.offsetX;
+			this.y = mouseY + this.offsetY;
+		}
+		if(this.x < 0)
+			this.x = 0
+		if(this.y < 0)
+			this.y = 0
+		if(this.x > windowWidth)
+			this.x = windowWidth
+		if(this.y > windowHeight)
+			this.y = windowHeight
+	}
+
+	draw(){
+		stroke(0);
+		if (this.dragging) {
+			fill(50);
+		} else if (this.rollover) {
+			fill(100);
+		} else {
+			fill(175, 200);
+		}
+		ellipse(this.x, this.y, this.w, this.h);
+	}
+
+	pressed(){
+		if (mouseX > this.x && mouseX < this.x + this.w && mouseY > this.y && mouseY < this.y + this.h) {
+			this.dragging = true;
+			this.offsetX = this.x - mouseX;
+			this.offsetY = this.y - mouseY;
+		}
+	}
+
+	released(){
+		this.dragging = false;
+	}
+}
+
+class Ruler {
+
+	constructor(){
+		this.mainx = windowWidth/2;
+		this.mainy = windowHeight/2;
+		this.shape1 = new Draggable(this.mainx - (this.mainx/2), this.mainy - (this.mainy/2), 10, 10)
+		this.shape2 = new Draggable(this.mainx + (this.mainx/2), this.mainy + (this.mainy/2), 10, 10)
+	}
+
+	draw(){
+		this.shape1.over()
+		this.shape1.update()
+		this.shape1.draw()
+		this.shape2.over()
+		this.shape2.update()
+		this.shape2.draw()
+
+		stroke(225)
+		line((this.shape1.x), (this.shape1.y), (this.shape2.x), (this.shape2.y));
+		rect((this.shape1.x + this.shape2.x)/2,(this.shape1.y + this.shape2.y)/2-20, 50, 20)
+		let dist = (parseInt(sqrt(pow(this.shape1.x - this.shape2.x,2)+pow(this.shape1.y - this.shape2.y,2))))
+		fill(0)
+		textSize(16)
+		text(String(dist),(this.shape1.x + this.shape2.x)/2-15,(this.shape1.y + this.shape2.y)/2-15)
+	}
+}
+
+
 let allObjects;
 let isMouseBeingPressed = false;
 let paused = false;
-let canvas;
+let canvas, r;
```

Comment:
Could we use a more descriptive name? Probably `ruler`?

Reply:
Oops, been needing to conserve characters for a programming game recently and happened to condense the variable name. I'll fix these few changes real quick.

Reply:
No worries!